arithmetization/src/main/java/net/consensys/linea/zktracer/module/add/AddOperationComparator.java

```diff
@@ -0,0 +1,35 @@
+/*
```

Comment:
It seems odd that you've implemented all the `Comparators` separately from the class itself.  You can have e.g. done `class AddOperation extends ModuleOperation implements Comparable<AddOperation>` then added a `compareTo()` method as required for the interface.  Just saves lines I guess --- not a big deal though.

Reply:
I suppose the reason for doing it this way is to avoid implemented a `Comparator` for all `Operation` classes?  I think you still could do it though by modifying the signature of `OperationSetModule.sortOperations()`.  I'd have to check it worked though --- so not worth bothering!

Reply:
Still, one reason for having `AddOperation` implement `Comparable<AddOperation>` is that you can then modify `ModuleOperationStackedSet` to require that its elements must be sortable like so:

```java
public class ModuleOperationStackedSet<E extends ModuleOperation & Comparable<E>>
    extends StackedSet<E> {
```

That just helps reduce the chance of a problem.

Reply:
> There is some inconsistency, in that e.g. BlockHash does things differently. But perhaps it has to?

made more consistent. For BlockHash we have to sort before commit time, as we have to trigger WCP.

>Does ExpOperation need the annotation @EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = false) 

added